repos/spack_repo/builtin/packages/pbvr/package.py

```diff
@@ -0,0 +1,148 @@
+        compiler_name = self.spec.compiler.name
+        if compiler_name != "gcc":
+            raise InstallError(f"{self.name} is only supported with the GCC compiler.")
+
+    def patch(self):
+        source_dir = self.stage.source_path
+        for root, dirs, files in os.walk(source_dir):
+            for fname in files:
+                path = os.path.join(root, fname)
+                try:
+                    with open(path, "r", encoding="utf-8") as f:
+                        content = f.read()
+                    if "KVS_DIR" in content:
+                        filter_file("KVS_DIR", "SPACK_KVS_DIR", path)
+                except Exception:
+                    pass
+
+    def build(self, spec, prefix):
+        # Workaround for qmake's $$system() not capturing command output correctly.
+        # This issue is caused by a known bug in certain Linux kernel versions.
+        # To avoid it, Qt is built with the '-no-feature-forkfd_pidfd' option to disable
+        # the use of new process management features that rely on pidfd.
+        # Note: Red Hat fixed this kernel issue in version 4.18.0-392 and later.
+        is_rhel8_bug_fixed = False
+        release_str = platform.uname().release
+        match = re.match(r"^(\d+\.\d+\.\d+)-([\d\.]+)\.el8", release_str)
+        if match and not is_rhel8_bug_fixed:
+            base_version = match.group(1)
+            build_number = match.group(2).split(".")[0]
+            full_version = f"{base_version}.{build_number}"
+            if Version(full_version) < Version("4.18.0.392"):
+                raise InstallError(
+                    f"The kernel version of this system is ${full_version}.\n"
+                    "You get an error when running qmake after installing qt-base"
+                    " on Red Hat Enterprise Linux 8 versions older than 8.7.\n"
+                    "You need to fix the package.py file for qt-base,"
+                    " so please refer to the following URL.\n"
+                    "https://github.com/CCSEPBVR/CS-IS-PBVR/wiki/BuildforLinux_JP"
+                )
+
+        with set_env(
+            KVS_CPP="g++",
+            SPACK_KVS_DIR=prefix,
+            VTK_VERSION="9.3",
+            VTK_INCLUDE_PATH=str(spec["vtk"].prefix.include) + "/vtk-9.3",
+            VTK_LIB_PATH=str(spec["vtk"].prefix.lib),
+        ):
+            # Build KVS
+            build_dir = join_path(self.stage.source_path, "KVS")
+            with working_dir(build_dir):
+                make()
+                make("install")
+
+            # Build Client
+            if "+client" in spec:
+                qmake = Executable(spec["qt-base"].prefix.bin.qmake)
+                build_dir = join_path(self.stage.source_path, "Client/build")
+                os.makedirs(build_dir)
+                with working_dir(build_dir):
+                    qmake("../pbvr_client.pro")
+                    make()
+
+            # Build Sevrer
+            make("-C", "CS_server")
```

Comment:
For consistency and to avoid the dependency on the present working directory:
```suggestion
            with working_dir(join_path(self.stage.source_path, "CS_server")):
                make()
```